main.go

```diff
@@ -117,30 +122,134 @@ type ObjRef struct {
 	Offloaded uint
 }
 
-// updateAutoretrieveIndex ticks every tickInterval and checks for new information to add to autoretrieve
-// If so, it updates the filecoin index with the new CIDs, saying they are present on autoretrieve
+// updateAutoretrieveIndex ticks every tickInterval and checks for new CIDs added
+// It updates storetheindex with the new CIDs, saying they are present on autoretrieve
 // With that, clients using bitswap can query autoretrieve servers using bitswap and get data from estuary
 func (s *Server) updateAutoretrieveIndex(tickInterval time.Duration, quit chan struct{}) error {
-	var autoretrieves []Autoretrieve
+	var autoretrieves []autoretrieve.Autoretrieve
 	var lastTickTime time.Time
-	ticker := time.NewTicker(tickInterval)
+	var curTime time.Time
+	var newContextID []byte
+	var err error
+
+	// Create index-provider engine (s.Node.IndexProvider) to send announcements to
+	// this needs to keep running continuously because storetheindex
+	// will come to fetch for advertisements "when it feels like it"
+	s.Node.ArEngine, err = autoretrieve.NewAutoretrieveEngine()
+	if err != nil {
+		return err
+	}
 
+	s.Node.ArEngine.RegisterMultihashLister(func(ctx context.Context, contextID []byte) (provider.MultihashIterator, error) {
```

Comment:
move this registration into the Autoretrieve package - register it as part of NewAutoretrieveEngine internally, no need to expose it here

Reply:
so in `main.go` we can have something;

```
arMgr, err  := autoretrieve.NewAutoretrieveEngine() // this should register the listener and also call start - this way main can fail if the manger/engine cannot be created/started
if err != nil {
	return err
}
defer arMgr.Shutdown()
go arMgr.Run(context.Background()) // move everything in https://github.com/application-research/estuary/pull/159/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R182 here

Reply:
this main, main will be lite and autoretrieve APIs will be easy to see how it can be used

Reply:
I thought of doing something along those lines but on `NewAutoretrieveEngine` when registering the lister we need to be able to access `s` to do `s.DB.Find`, and I'm not sure how to do this on this model

Reply:
then pass the `s.DB` to `NewAutoretrieveEngine` and set a DB field in the engine